File: src/main/java/org/labun/jooq/generator/task/TaskContext.java
package org.labun.jooq.generator.task;

import lombok.Getter;
import lombok.Setter;
import org.labun.jooq.generator.GeneratorContext;
import org.labun.jooq.generator.config.CodeGenerationConfig;
import org.labun.jooq.generator.util.FilePathResolver;
import org.labun.jooq.generator.util.NameCreator;
import org.labun.jooq.generator.util.TypeResolver;

/**
 * @author Konstantin Labun
 */
@Getter
@Setter
public class TaskContext {
    private GeneratorContext generatorContext;
    private CodeGenerationConfig config;
    private NameCreator nameCreator;
    private TypeResolver sqlTypeResolver;
    private TypeResolver javaTypeResolver;
    private FilePathResolver filePathResolver;
}

File: src/main/kotlin/org/labun/jooq/generator/task/TaskContext.kt
package org.labun.jooq.generator.task

import lombok.Getter
import lombok.Setter
import org.labun.jooq.generator.GeneratorContext
import org.labun.jooq.generator.config.CodeGenerationConfig

/*
 * @author Konstantin Labun
 */
@Getter
@Setter
class TaskContext {

    private val generatorContext: GeneratorContext? = null
    private val config: CodeGenerationConfig? = null
    private val nameCreator: NameCreator? = null
    private val sqlTypeResolver: TypeResolver? = null
    private val javaTypeResolver: TypeResolver? = null
    private val filePathResolver: FilePathResolver? = null
}
